### ISSUE TITLE
`:DiffviewFileHistory` fails with `too many results to unpack`

### ISSUE THREAD

**Issue by matu3ba** [sindrets/diffview.nvim#155]
## Problem

```
Error executing vim.schedule lua callback: ...m/lua/diffview/views/file_history/file_history_panel.lua:169: too many results to unpack
stack traceback:
        [C]: in function 'unpack'
        ...m/lua/diffview/views/file_history/file_history_panel.lua:169: in function 'update_components'
        ...m/lua/diffview/views/file_history/file_history_panel.lua:228: in function <...m/lua/diffview/views/file_history/file_history_panel
.lua:206>
```

## Solutions

1. Set a timeout (configurable)
2. Set max results to unpack (configurable)
3. Set max amount of commits backwards from HEAD (configurable)

## Minimal solution (user should set it properly, so only provide error diagnostics)

Error should print 1. the amounts of results that were unpacked when the error/failure occurred, 2. time, 3. commit range backwards from HEAD, 4. commit on which it failed unpacking

**Comment by matu3ba** [comment]
Sidenode: A simple way to stop/kill the "current" indexing process (+ close the windows, if it would otherwise create inconsistent state) would be helpful.

**Comment by sindrets** [comment]
> Sidenode: A simple way to stop/kill the "current" indexing process (+ close the windows, if it would otherwise create inconsistent state) would be helpful.

@matu3ba solved in ca68e1c78a3aefe6f1f029903df10e5592e7ba9d.

**Comment by matu3ba** [comment]
awesome, thanks alot!
